client/app/deskaas/deskaas-confirm-terminate/deskaas-confirm-terminate.html

```diff
@@ -1,68 +1,51 @@
-                    </div>
-                </div>
-                <div class="oui-field">
-                    <label for="vdi-comment" class="oui-field__label oui-label" data-translate="vdi_confirm_cancel_commentary"></label>
-                    <div class="oui-field__content">
-                        <div class="oui-field-control oui-field-control_4">
-                            <input
-                                type="text"
-                                id="vdi-comment"
-                                class="oui-input"
-                                data-ng-model="DeskaasConfirmTerminateCtrl.values.commentary"
-                                value="DeskaasDetailsCtrl.stateParams.token"
-                                required />
-                        </div>
-                    </div>
-                </div>
-            </div>
-        </cui-modal-body>
-        <cui-modal-footer>
-            <cui-cancel-button data-text="'common_modal_cancel' | translate"
-                data-on-click="DeskaasConfirmTerminateCtrl.cancel()"></cui-cancel-button>
-            <cui-submit-button data-text="'common_modal_ok' | translate"
-                data-disabled="DeskaasConfirmTerminateCtrl.flags.init || !(DeskaasConfirmTerminateCtrl.values.token && DeskaasConfirmTerminateCtrl.values.reason) || !confirmTerminateForm.$valid"
-                data-on-click="DeskaasConfirmTerminateCtrl.ok()"></cui-submit-button>
-        </cui-modal-footer>
-    </form>
-</cui-modal>
+<form id="confirmTerminateForm"
+      name="confirmTerminateForm"
+      data-ng-submit="confirmTerminateForm.$valid && DeskaasConfirmTerminateCtrl.ok()"
+      novalidate>
+    <oui-modal
+        data-secondary-action="$ctrl.cancel()"
+        data-title="{{ ::'vdi_cancel_title' | translate }}"        
+        data-secondary-label="{{ ::'common_modal_cancel' | translate }}"
+        data-primary-label="{{ ::'common_modal_ok' | translate}}"
+        data-on-dismiss="$ctrl.cancel()"
+        data-loading="DeskaasConfirmTerminateCtrl.flags.init">
+        <oui-field data-label="{{ ::'Token' | translate }}">
+            <input
+                data-type="text"
+                name="vdi-token"
+                id="vdi-token"
+                class="oui-input"
+                data-ng-model="DeskaasConfirmTerminateCtrl.values.token"
+                data-ng-disabled="true"
+                required />
+        </oui-field>
+        <oui-field data-label="{{ ::'vdi_confirm_cancel_reason' | translate }}" data-size="m">
+            <label class="oui-select">
+                <select id="vdi-reason"
+                    name="vdi-reason"
+                    data-ng-model="DeskaasConfirmTerminateCtrl.values.reason"
+                    class="oui-select__input"
+                    required>
+                    <option selected value="" data-translate="common_select"></option>
+                    <option value="TOO_HARD_TO_USE" data-translate="vdi_reason_hard">
```

Comment:
Why not use ngOptions? 

This list should be return by API I think :)

(it's an enum from API)

Reply:
It's strange because we are developing a similar and generic form in dedicated part

Reply:
Exemple here: https://github.com/ovh-ux/ovh-manager-dedicated/pull/415/files

Reply:
Yes the validation criteria is coming from an API. And we can make the validation dynamic too. But need to work on this in a new ticket.